pypeit/flatfield.py

```diff
@@ -687,8 +687,16 @@ def fit(self, spat_illum_only=False, debug=False):
         # Model each slit independently
         for slit_idx, slit_spat in enumerate(self.slits.spat_id):
             # Is this a good slit??
-            if self.slits.mask[slit_idx] != 0:
+            if self.flatpar['pixelflat_min_wave'] is not None or self.flatpar['pixelflat_max_wave'] is not None:
+                # In this case we consider good 'BOXSLIT' (bpm=2)
+                good_bpms = [0, 2]
+            else:
+                # In this case we consider good 'BOXSLIT' (bpm=2), bad wavecalib (bpm=8)
+                # and the combination of the 2 (bpm=10)
+                good_bpms = [0, 2, 8, 10]
```

Comment:
I'm not quite following the logic here and what these good_bpms mean. Rather than checking 8 and 10, I think you should be using the bit arithmetic routines that @kbwestfall developed to check for the relevant bits. The code would also be clearer and more robust if you operated on the bit names rather than the bit integers, since that mapping could change. Finally, I don't get why a BOXSLIT is now a good_bpm whereas it was not before. We don't flat field the box slits?